ludwig/config_validation/checks.py

```diff
@@ -0,0 +1,327 @@
+                f"'{input_feature[TYPE]}'. Input feature types: {list(get_encoder_registry().keys())}."
+            )
+        if ENCODER in input_feature:
+            if (
+                TYPE in input_feature[ENCODER]
+                and input_feature[ENCODER][TYPE] not in get_encoder_registry()[input_feature[TYPE]]
+            ):
+                raise ConfigValidationError(
+                    f"Encoder type '{input_feature[ENCODER][TYPE]}' for input feature {input_feature[NAME]} must be "
+                    f"one of: {list(get_encoder_registry()[input_feature[TYPE]].keys())}."
+                )
+
+    # Check output features.
+    for output_feature in config[OUTPUT_FEATURES]:
+        if NAME not in output_feature:
+            raise ConfigValidationError("All output features must have a name.")
+        if TYPE not in output_feature:
+            raise ConfigValidationError(f"Output feature {output_feature[NAME]} must have a type.")
+        if output_feature[TYPE] not in get_decoder_registry():
+            raise ConfigValidationError(
+                f"Output feature {output_feature[NAME]} uses an invalid/unsupported output type "
+                f"'{output_feature[TYPE]}'. Supported output features: {list(get_decoder_registry().keys())}."
+            )
+        if DECODER in output_feature:
+            if (
+                TYPE in output_feature[DECODER]
+                and output_feature[DECODER][TYPE] not in get_decoder_registry()[output_feature[TYPE]]
+            ):
+                raise ConfigValidationError(
+                    f"Decoder type for output feature {output_feature[NAME]} must be one of: "
+                    f"{list(get_decoder_registry()[output_feature[TYPE]].keys())}."
+                )
+
+    # Check combiners.
+    if config.get(MODEL_TYPE, MODEL_ECD) == MODEL_ECD:
+        if COMBINER not in config:
+            return
+        if TYPE not in config[COMBINER]:
+            raise ConfigValidationError("Combiner must have a type.")
+        if config[COMBINER][TYPE] not in get_combiner_registry():
+            raise ConfigValidationError(f"Combiner type must be one of: {list(get_combiner_registry().keys())}.")
+
+    # Check trainer.
+    if TRAINER in config and config[TRAINER] is None:
+        raise ConfigValidationError("Trainer cannot be None.")
+
+    # Check optimizer.
+    if TRAINER in config and "optimizer" in config[TRAINER]:
+        if config[TRAINER]["optimizer"] is None:
+            raise ConfigValidationError("Trainer.optimizer cannot be None.")
+        if TYPE in config[TRAINER]["optimizer"]:
+            if config[TRAINER]["optimizer"][TYPE] not in optimizer_registry:
+                raise ConfigValidationError(
+                    f"Trainer.optimizer.type must be one of: {list(optimizer_registry.keys())}."
+                )
+
+
+@register_config_check
+def check_feature_names_unique(config: ModelConfigDict) -> None:
```

Comment:
I would remove this one since it's in the schema already and I would prefer to keep a single source of turth ofr this.

Reply:
@tgaddair Where in the schema? If you are referring to `uniqueItemProperties` remember that that's not an official JSON keyword - merely an extra one that AJV supports. So we would get the validation for unique feature names in the FE but not on the Ludwig side if we removed this check.

Reply:
Yep, good call. I think it's fine to keep this in, then.